core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -242,6 +242,42 @@ case class PSBT(
     PSBT(globalMap, newInputMaps, outputMaps)
   }
 
+  def addScriptWitnessToInput(
+      scriptWitness: ScriptWitness,
+      index: Int): PSBT = {
+    require(
+      index < inputMaps.size,
+      s"index must be less than the number of input maps present in the psbt, $index >= ${inputMaps.size}")
+    require(!inputMaps(index).isFinalized,
+            s"Cannot update an InputPSBTMap that is finalized, index: $index")
+
+    val previousElements = inputMaps(index).elements
+      .filterNot(_.key.head == PSBTInputKeyId.WitnessScriptKeyId.byte)
+
+    val newMap = scriptWitness match {
+      case p2wpkh: P2WPKHWitnessV0 =>
+        val newElement =
+          InputPSBTRecord.WitnessScript(P2PKHScriptPubKey(p2wpkh.pubKey))
```

Comment:
I was under the impression that no `WitnessScript` record was needed for `p2wpkh`, did you find something suggesting otherwise? If so other things will have to change as well as we derive it in different ways elsewhere.

Reply:
Okay, I see below that it gets removed but don't understand why you add it just to remove it

Reply:
So it knows that it is a witness utxo so it changes from a `NonWitnessOrUnknownUTXO` to a `WitnessUTXO`

Reply:
Nowhere in the code below for compressing the map does it look at the `WitnessScript` field, it exclusively looks at the `ScriptPubKey`. Still don't see why we add and remove this `newElement`

Reply:
Nowhere in the code below for compressing the map does it look at the `WitnessScript` field, it exclusively looks at the `ScriptPubKey`. Still don't see why we add and remove this `newElement`

Reply:
My bad was thinking we used `WitnessScript` but we actually use `WitnessScript`